### DISCUSSION TITLE
Improve discounts integration with cart transformations

### DISCUSSION THREAD

**Discussion by a47ae** [Shopify/function-examples#532] [Feedback: Cart Transform API]
We're currently using a cart transform `expand operation` to add fees to cart items in Shopify. This operation expands each item into two components: the original item and the associated fee. However, we've encountered an issue with how discounts are applied in this scenario.

**Current Behavior**
Discounts are only applied to the bundle parent product, affecting the combined price (item + fee) instead of just the item price.

Example:
We have a line item in the cart with `Product A`. We then expand this into `Product A` and `Fee B`. No if the customer applies a discount for 10 %, valid for `Product A`, the following calculation happens

```
(`Product A - price` + `Fee B - price`) * 0,9
```

our desired Behavior would be 

```
(`Product A - price` * 0,9) + `Fee B - price`
```

**The problem could potentially be solved by:**

- Gaining access to discounts which will be applied to a line item within the cart transform function
  - This would allow us to increase the price of the fee to compensate the discount
- Implementing logic for discounts to operate only on bundle children
  - Probably this is out of scope for the functions team, but it would be neat to allow merchants to apply discounts on a bundle component basis
- Changing the order of operations (applying discounts before cart transformation)
  - If the discounts would be applied before the cart transformation this could also be solved

**Comment by andrewchen5678** [comment]
Hit the same issue today
